compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -2026,6 +2026,10 @@ static LogicalResult adjustTileSizesForPackOp(func::FuncOp entryPointFn,
     if (hasChanged)
       return WalkResult::interrupt();
 
+    // Skip the adjustment if the pack op is not one of consumer ops.
+    if (packOp->isBeforeInBlock(rootOp))
```

Comment:
Could we use the ssa use-def chains instead of looking for next position in block?

Reply:
It is hard. In practice case, we don't really see the case. It only happens when we have scalar program. In this context, we fuse everything into a single dispatch. That's why we need to handle `pack + mmt4d + unpack` here. (In the past, they are dynamic shapes, so they won't get fused.)

Reply:
Chatted offline about a better way of doing this.